arches_lingo/src/arches_lingo/components/generic/LabelEditor.vue

```diff
@@ -0,0 +1,141 @@
+<script setup lang="ts">
+import { defineProps, onMounted, ref } from "vue";
+import { fetchControlledListOptions } from "@/arches_lingo/api.ts";
+
+import NonLocalizedString from "@/arches_lingo/components/generic/NonLocalizedString.vue";
+import ReferenceDatatype from "@/arches_lingo/components/generic/ReferenceDatatype.vue";
+// import ResourceInstanceRelationships from "@/arches_lingo/components/generic/ResourceInstanceRelationships.vue";
+
+import {
+    EDIT,
+    LANGUAGE_CONTROLLED_LIST,
+    LABEL_CONTROLLED_LIST,
+    STATUSES_CONTROLLED_LIST,
+    METATYPES_CONTROLLED_LIST,
+    EVENT_TYPES_CONTROLLED_LIST,
+} from "@/arches_lingo/constants.ts";
+
+import type {
+    AppellativeStatus,
+    ControlledListItem,
+    ControlledListItemResult,
+} from "@/arches_lingo/types.ts";
+
+const props = withDefaults(
+    defineProps<{
+        value?: AppellativeStatus;
+    }>(),
+    {
+        value: () => ({}) as AppellativeStatus,
+    },
+);
+const appellative_status = ref(props.value);
+
+function onUpdate(newValue: string) {
+    console.log(newValue);
+}
+
+async function getOptions(listId: string): Promise<ControlledListItem[]> {
```

Comment:
nit: functions should be declared last, after all consts ( I def need to get that styleguide updated 😅 )